tensorflow/core/platform/ctstring_internal.h

```diff
@@ -162,6 +162,14 @@ static inline size_t TF_TString_ToInternalSizeT(size_t size,
 #else   // TF_TSTRING_LITTLE_ENDIAN
   // 0xFF000000 or 0xFF00000000000000 depending on platform
   static const size_t mask = ~((~(size_t)0) >> 8);
+#if defined(__s390x__)
+  if (type == TF_TSTR_OFFSET) {
+    return (mask & (size << 2)) | (~mask & (size << 2)) |
```

Comment:
This doesn't look right... I think the `type` information will be merged with existing (non-masked) bits.

Let's say the`size` is `0x00000000 3FFFFFFF`, and the type is `0x02` (offset)
```
size = 0x00000000 3FFFFFFF
mask = 0xFF000000 00000000

 mask & (size << 2) = 0x00000000 00000000
~mask & (size << 2) = 0x00000000 FFFFFFFC
         type << 24 = 0x00000000 20000000
                   _______________________
or                  = 0x00000000 FFFFFFFC
```
The type information is then corrupted.  I believe you want the mask to always be `0xFF000000` in the case of offset.